flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/ExactlyOnceKafkaWriter.java

```diff
@@ -293,22 +309,38 @@ private TransactionAbortStrategyContextImpl getTransactionAbortStrategyContext(
                     producerPool.recycle(producer);
                     return epoch;
                 };
-        Set<String> ongoingTransactionIds =
+        Set<String> precommittedTransactionalIds =
                 recoveredStates.stream()
                         .flatMap(
                                 s ->
-                                        s.getOngoingTransactions().stream()
+                                        s.getPrecommittedTransactionalIds().stream()
                                                 .map(CheckpointTransaction::getTransactionalId))
                         .collect(Collectors.toSet());
+        Set<Integer> oldSubtaskIds =
```

Comment:
Nit: I wonder whether the `getTransactionAbortStrategyContext` method isn't better placed inside the `TransactionAbortStrategyContextImpl` as a builder method or similar